### PR TITLE
Remove dead upgrade-mode 'U' option to postgres and postmaster.

### PR DESCRIPTION
**Pull request by ashwinstar**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`

### CODE REVIEW

File: src/backend/tcop/postgres.c [hunk 1]
```diff
@@ -4439,18 +4439,6 @@ process_postgres_switches(int argc, char *argv[], GucContext ctx,
...
 				}
 
-			case 'U':
-				/*
-				 * In upgrade mode, we indicate we're in upgrade mode and
-				 * 1. allow DML on persistent table & catalog table
-				 * 2. alter DDL on catalog table (NOTE: upgrade_mode must set beforehand)
-				 * 3. TODO: disable the 4.1 xlog format (stick with the old)
-				 */
-				SetConfigOption("upgrade_mode",                         "true", ctx, gucsource);
```
Comment (danielgustafsson) [inline src/backend/tcop/postgres.c]: The associated GUCs `gp_upgrade_mode` (which is entirely dead), `gp_maintenance_mode` and `gp_maintenance_conn` (which are mostly dead) should also be removed with this.

Comment (ashwinstar) [reply src/backend/tcop/postgres.c]: `gp_maintenance_mode` and `gp_maintenance_conn` is associated with `-m` maintenance_mode option which is different. And I feel that serves purpose, to only allow super user to connect. Hence, seems still valid option and hence guc needs to remain.

Will delete the extern remaining for `gp_upgrade_mode`.

**Comment by danielgustafsson** [comment]
> On 4 Jan 2019, at 22:52, Ashwin Agrawal <notifications@github.com> wrote:

> gp_maintenance_mode and gp_maintenance_conn is associated with -m maintenance_mode option which is different.

Correct, I was grepping in the wrong place and missed that they had actual usage in the code.
> Will delete the extern remaining for gp_upgrade_mode.
> 
Thanks

**Comment by ashwinstar** [comment]
Merged https://github.com/greenplum-db/gpdb/commit/a52fa413c2982cf61bcb950c4b90aa8b341d22d9

**Pull request closed without merging**